### ISSUE TITLE
[Bug]: Permission.locationWhileInUse.request() returns granted when user chooses 'Allow only onece'

### ISSUE THREAD

**Issue by Smibser** [Baseflow/flutter-permission-handler#1128]
### Please check the following before submitting a new issue.

- [X] I have searched the [existing issues](https://github.com/baseflow/flutter-permission-handler/issues).
- [X] I have carefully [read the documentation](https://github.com/Baseflow/flutter-permission-handler/blob/main/permission_handler/README.md) and verified I have added the required platform specific configuration.

### Please select affected platform(s)

- [ ] Android
- [X] iOS
- [ ] Windows

### Steps to reproduce

1. request locationWhileInUse
2. Choose 'Allow Only Once' on the ios device

### Expected results

I expected to return denied since location while in use is not granted. I also found this PR https://github.com/Baseflow/flutter-permission-handler/pull/1082 which it should have fixed. But I am still facing this issue

### Actual results

It returns PermissionStatus.granted. This let my code request always permission but that's not possible so it fails

### Code sample

<details><summary>Code sample</summary>

```dart
final locationWhenInUseStatus = await Permission.locationWhenInUse.request();
    print("Location when in use status: $locationWhenInUseStatus");
    if (locationWhenInUseStatus.isGranted) {
      print("Location when in use is granted");
      var locationAlwaysStatus = await Permission.locationAlways.request();
      print("After requesting location always");

      if (!locationAlwaysStatus.isGranted) {
        print("Location always is denied");
        bool res = await openAppSettings();
      }
      
    }
    if (!locationWhenInUseStatus.isGranted) {
      print("Location when in use is not granted");
    }
```

</details>

### Version

10.2.0

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.7.12, on macOS 13.3.1 22E772610a darwin-arm64, locale en-DE)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.2)
[✓] IntelliJ IDEA Ultimate Edition (version 2020.3.2)
[✓] VS Code (version 1.81.1)
[✓] Connected device (4 available)
[✓] HTTP Host Availability

• No issues found!
```

</details>

**Comment by TimHoogstrate** [comment]
Dear @Smibser,

On iOS the    
```
case PermissionGroupLocation:
case PermissionGroupLocationAlways:
case PermissionGroupLocationWhenInUse:
```
Are basically the same permissions. So if you request one of them you'll see the same Alert. If you permanently deny, you cannot request the permissions a second time. (The user has to change the settings in the AppSettings). If you allow once,  and if you kill the app, you will be able to request permissions again. If you allow once, the permission status of locationAlways will be of status permanentlyDenied. (until you kill the app and run it again).

In short, this is expected behaviour so I'll close the issue for now.

Kind regards,

**Comment by Smibser** [comment]
Okay, now I got it.
But when I grant 'WhileInUse' on the first dialog I also get permanentlyDenied as permission status of locationAlways. But in that case I can request locationAlways
